WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -363,6 +363,10 @@ - (ReaderTopic *)siteTopicForPost:(ReaderPost *)post
     topic.topicDescription = post.blogDescription;
     topic.path = path;
 
+    [self.managedObjectContext performBlockAndWait:^{
```

Comment:
I'm wrestling a bit with this. I think wrapping the insides of the method in a performBlockAndWait would be weird since we're returning a value. It also seems bad to make the caller responsible for only ever calling the method with a perform block.
I think what we need to do is refactor to use performBlockAndWait internally, and instead of returning a topic, use a completion handler on the main queue.  But maybe for the purpose of this patch we use the NSAssert? 
Reader topics are being violently refactored very soon so I'm not too keen on doing too much more here. 
Thoughts?